packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/invoke-model.ts

```diff
@@ -211,7 +257,36 @@ export class BedrockInvokeModel extends sfn.TaskStateBase {
         Output: this.props.output?.s3Location ? {
           S3Uri: `s3://${this.props.output.s3Location.bucketName}/${this.props.output.s3Location.objectKey}`,
         } : undefined,
+        GuardrailIdentifier: this.props.guardrailConfiguration?.guardrailIdentifier,
+        GuardrailVersion: this.props.guardrailConfiguration?.guardrailVersion,
+        Trace: this.props.traceEnabled === undefined
+          ? undefined
+          : this.props.traceEnabled
+            ? 'ENABLED'
+            : 'DISABLED',
       }),
     };
   }
+
+  private validateGuardrailConfiguration(props: BedrockInvokeModelProps) {
+    if (!props.guardrailConfiguration) return;
+
+    const { guardrailIdentifier, guardrailVersion } = props.guardrailConfiguration;
+
+    if (!Token.isUnresolved(guardrailIdentifier)) {
+      const guardrailConfigurationPattern = /^(([a-z0-9]+)|(arn:aws(-[^:]+)?:bedrock:[a-z0-9-]{1,20}:[0-9]{12}:guardrail\/[a-z0-9]+))$/;
+      if (!guardrailConfigurationPattern.test(guardrailIdentifier)) {
+        throw new Error(`guardrailIdentifier must match the ^(([a-z0-9]+)|(arn:aws(-[^:]+)?:bedrock:[a-z0-9-]{1,20}:[0-9]{12}:guardrail\/[a-z0-9]+))$ pattern, got ${guardrailIdentifier}`);
```

Comment:
Could we also add validation on the [length constraints](https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_InvokeModel.html#API_runtime_InvokeModel_RequestSyntax:~:text=Length%20Constraints%3A%20Minimum%20length%20of%200.%20Maximum%20length%20of%202048.)? 

Min. 0, Max. 2048.

Reply:
@paulhcsun 
Thanks. 
I've added a validation and a unit test.

Reply:
If you are going to provide this kind of error checking could you please make the error message descriptive of what the expected input is instead of regex? It's better readability.

Reply:
@TheRealAmazonKendra 
Thanks. I've changed the error message.